packages/resolver/lib/sources/globalnpm.ts

```diff
@@ -14,7 +14,7 @@ export class GlobalNPM implements ResolverSource {
 
     let [packageName] = importPath.split("/", 1);
     if (detectInstalled.sync(packageName)) {
-      const regex = new RegExp(`/${packageName}$`);
+      const regex = new RegExp(`${path.sep}${packageName}$`);
```

Comment:
Careful here!  I don't think this does what you want.

Inputs to the `RegExp` constructor take as input a regexp string.  If it contains regexp-significant characters -- which a backslash certainly is! -- the result won't be what you want.

In order to solve this, you either need to escape the string before you pass it in (lodash has a function for this, that we use in some places), or -- and I think this is the better option when possible -- not use a regex.

Fortunately, it looks like in this case we can indeed get away with not using a regex.  It looks to me like this regex is only being used as the first argument to `replace`.  You don't need to pass a regex there, you can pass a plain string and it'll be interpreted as a plain string.  So, that's the thing to do here.

(Note that if it were a *global* regex, then you'd be forced to take the escaping route; we can't use `replaceAll` because that currently only exists on Node 16.  Fortunately it isn't.)

Reply:
Good catch! I missed the \ metachar here! The original used a regex to match the suffix and a non-regex could potentially match a non-suffix portion of the string. What do you think of  ```RegExp(`\\${path.sep}${packageName}$`)```?

Reply:
Oh, I see, I missed the part that you want to match specifically a suffix.

Your version might work, but it's not obvious to me; can you check whether that sort of `RegExp(...)` does indeed work as you expect?  If it does then great!  Although you'll want to leave an explanatory comment for sure.

If it doesn't you might want to use a proper escaping package (again, lodash has that), or find a non-regex way of doing it (there are certainly ways... like use `endsWith` to see if it matches, and, if it does, slice off an appropriate number of characters and append the new thing -- obviously that's a bit more involved).

Reply:
OK I just checked it myself and it seems that your approach does indeed work, regarding `path.sep`.  That said, I still think it might be a good idea to either use a proper escaping library, or to take the non-regex route, on account of the possibility of special characters in `packageName` -- unless there's some reason we can be sure that won't happen?  I think NPM packages might be allowed to include `.` in their names, which might cause problems.

Reply:
For the `.`, it would be ok, a dot meta will match the dot target. But, I'll sleep better using `endsWith`. Thanks for getting me here @haltman-at